includes/feeds/google.php

```diff
@@ -556,12 +557,25 @@ public function make_request($id = '', $time_min = 0, $time_max = 0)
 			}
 
 			$is_authhelper = get_option('simple_calendar_run_oauth_helper');
+			$feed_type = wp_get_object_terms($post_id, 'calendar_feed');
+
 			// Query events in calendar.
 			$simple_calendar_auth_site_token = get_option('simple_calendar_auth_site_token');
 			$response = '';
-			if (isset($simple_calendar_auth_site_token) && !empty($simple_calendar_auth_site_token && $is_authhelper)) {
-				$response = apply_filters('simple_calendar_oauth_list_events', '', $id, $args);
-
+			if (
+				isset($simple_calendar_auth_site_token) &&
+				!empty($simple_calendar_auth_site_token && $is_authhelper) &&
+				isset($feed_type[0]->slug) &&
+				$feed_type[0]->slug != 'google'
+			) {
+				$backgroundcolor = '';
+				$response_arr = apply_filters('simple_calendar_oauth_list_events', '', $id, $args);
+				
+				$response = unserialize($response_arr['data']);	
+				if(isset($response_arr['backgroundcolor']) && !empty($response_arr['backgroundcolor'])){			
+					$backgroundcolor = $response_arr['backgroundcolor'];
```

Comment:
Why are we writing all this logic in `make_request` function?

Reply:
Because make request is responsible to call Google for events, it does not matter if we are requesting public or private calendars event.